### PR TITLE
Fix legends color range update

### PR DESCRIPTION
**Pull request by elenatorro**
This PR fixes:

* Custom legend items order do not match the UI  https://github.com/CartoDB/support/issues/1761
* Heatmap legend does not update after style changes https://github.com/CartoDB/cartodb/issues/13763

*About the second issue: IMO, is quite confusing that the heatmap colors change when we change the colors in the editor, but it does not change the other parameters (such as the width, for instance) until we click on "Apply". Although his is the expected behaviour since it does not happen only for heatmaps, but for every aggregation style, I think this is not consistent*

### CODE REVIEW

**Comment by VictorVelarde** [comment]
@elenatorro I agree with you that heatmap behaviour should be consistent in issue #13763, updating just when expected. I'd double check with @ramiroaznar before applying this fix.

File: lib/assets/javascripts/builder/helpers/parser-css.js [hunk 1]
```diff
@@ -267,13 +267,33 @@ CartoParser.prototype = {
 
     if (opt && opt.mode === 'hex') {
       colors = _.map(colors, function (color) {
-        return cdb.Utils.rgbToHex(color[0], color[1], color[2]);
```
Comment (VictorVelarde) [inline lib/assets/javascripts/builder/helpers/parser-css.js]: Nice!

File: lib/assets/javascripts/builder/helpers/style.js [hunk 2]
```diff
@@ -46,6 +47,18 @@ module.exports = {
...
   },
 
+  getColorsForLegend: function (layerDefinitionModel) {
+    var changed = layerDefinitionModel.styleModel && layerDefinitionModel.styleModel.hasChanged();
+
+    if (changed) {
+      return this.getColorsFromRange(layerDefinitionModel.styleModel);
+    }
+
+    var content = layerDefinitionModel.cartocssModel.get('content');
```
Comment (VictorVelarde) [inline lib/assets/javascripts/builder/helpers/style.js]: Technically it looks fine, but after your comment, I have doubts if we should go ahead with this or we should remove the layer color update instead

File: lib/assets/javascripts/builder/components/form-components/editors/sortable-list.js [hunk 3]
```diff
@@ -39,8 +39,11 @@ Backbone.Form.editors.SortableList = Backbone.Form.editors.List.extend({
 
   _onSortableUpdate: function (event, ui) {
     var sorted = [];
-    _.each(this.items, function (item) {
-      var index = item.$el.index(this.$list);
+    var list = this.$list;
+    var items = this.items;
+
+    _.each(items, function (item) {
```
Comment (VictorVelarde) [inline lib/assets/javascripts/builder/components/form-components/editors/sortable-list.js]: Nice catch!

**Comment by ramiroaznar** [comment]
LGTM!

**Pull request merged**